crates/telio-dns/src/zone.rs

```diff
@@ -21,11 +21,22 @@ use crate::forward::ForwardAuthority;
 /// organization or administrator.
 pub(crate) type Zones = Catalog;
 
+/// Information stored by records
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum TelioRecord {
+    /// Only IPv4
+    OnlyIpv4(Ipv4Addr),
+    /// Only IPv6
+    OnlyIpv6(Ipv6Addr),
+    /// Both types of IP
+    Both(Ipv4Addr, Ipv6Addr),
+}
+
 /// Records (aka zone files) are instructions that live in authoritative
 /// DNS servers and provide information about a domain including what IP
 /// address is associated with that domain and how to handle requests
 /// for that domain.
-pub type Records = HashMap<String, (Option<Ipv4Addr>, Option<Ipv6Addr>)>;
+pub type Records = HashMap<String, TelioRecord>;
```

Comment:
Sorry for not spotting this in the previous review but is there a reason why we artificially restrict this type to have max two IPs for given domain? Normally you can get multiple IPv4 and IPv6 records so maybe instead of the `TelioRecord` this should be just `pub type Records = HashMap<String, Vec<std::net::IpAddr>>`? (Sadly there is no standard/common type that represents non empty vector).
This is just me thinking out loud, fell free to completely ignore if you think current approach is better.